src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -241,9 +241,10 @@ internal void CommitUnsynchronized()
             }
 
             // Always move the commit head to the write head
+            var bytesWritten = _currentWriteLength;
```

Comment:
Is it important for everything below runs if `_currentWriteLength == 0`? If not, why not just return here?